### DISCUSSION TITLE
Focus a window programmatically?

### DISCUSSION THREAD

**Discussion by m1027** [labwc/labwc#1992] [General]
Hi,

you might wonder why one would want to focus a window programmatically? ;-) Well, here is my use case:

I need to run a  browser in kiosk mode on a touch screen. Wayland / labwc is near to ideal for that.

However, web apps do behave quite differently if (for whatever reason) the browser window loses the focus. Two examples: Javascript code may get slowed down (it's a browser's feature), WebRTC sessions may not find their HW devices and such.

So, is there a way in labwc to

- find my window and
- programmatically activate it (focus it)

?

Thanks for any hints.

**Comment by Consolatis** [comment]
[wlrctl](https://git.sr.ht/~brocellous/wlrctl) should have you covered. It may be in your distro repo as well.

Edit:
You could maybe even use something like 
```sh
while true; do
    wlrctl toplevel waitfor firefox state:inactive && wlrctl toplevel focus firefox
done
```
